client/src/components/documents/documentView.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import '../../main.scss';
+import NavBar from '../navbar/navBar';
+import Dialog from '../diaLog/confirmDialog';
+import EditDocument from './documentEdit';
+import { currentUser } from '../../helpers/auth';
+import { fetchDocumentById, deleteDocument } from '../../actions/actionCreator';
+
+/**
+ *
+ * @class DocumentView
+ * @extends {Component}
+ */
+class DocumentView extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      loading: true
+    };
+    this.handleDocumentDelete = this.handleDocumentDelete.bind(this);
+  }
+  componentDidMount() {
```

Comment:
I think you can use `componentWillMount` here so that your data will be available before mount. Using component did mount will first render the component, reset the props and rerender the component.